src/coreclr/debug/ee/funceval.cpp

```diff
@@ -3483,7 +3483,7 @@ static void GCProtectArgsAndDoNormalFuncEval(DebuggerEval *pDE,
         ThrowHR(COR_E_OVERFLOW);
     }
     OBJECTREF * pObjectRefArray = (OBJECTREF*)_alloca(cbAllocSize);
-    *pObjectRefArray = {};
+    memset(pObjectRefArray, 0, cbAllocSize);
```

Comment:
```suggestion
    void* memObjRefArray = _alloca(cbAllocSize);
    memset(memObjRefArray, 0, cbAllocSize);
    OBJECTREF* pObjectRefArray = (OBJECTREF*)memObjRefArray ;
```

This is invoking undefined behavior in DEBUG/CHECKED builds. The one above also has an issue due to its fields having ctors. In DEBUG/CHECKED the `OBJECTREF` is a struct with a ctor and calling `memset` on a type with a ctor is undefined. I'm a bit surprised the `{}` wasn't zero initializing these types. Are we sure this was a case where memory wasn't zero'd out?

A simple "fix" is to `memset` the memory first and then cast.

Reply:
> I'm a bit surprised the {} wasn't zero initializing these types. Are we sure this was a case where memory wasn't zero'd out?

Ah. I missed that this logic is an array of X. This was clearly wrong, apologies.